### PR TITLE
Upgrades: Clean up lib/upgrades/constants

### PR DESCRIPTION
**Pull request by sirreal**
- Moves a heavily modified "constants" file. This helps expose usage, importing/requiring from a non-existent file should error. The file exported a single object containing action types. It made sense to rename this to `action-types.js`.
- Export named constant strings instead of the container object. Allows for better build optimization and helps to expose usage.
- Import names from the new file.  Allows for better build optimization and helps to expose usage.
- Prune unused names. This was done by hand, observing build warnings about importing names that aren't exported, thanks to the previous 2 points.

Look at commit 083dd85 to see the removed names.

Suggested by @tyxla (https://github.com/Automattic/wp-calypso/pull/25396#discussion_r194368006)

## Build impact

[This is a small improvement on bundle sizes:](http://iscalypsofastyet.com/branch?branch=remove/lib-upgrades/constants)

```
Commit: 38992e02c9a1624378736f4d9624503fb0aeb127
Author: sirreal
At: 2018-08-06T18:12:05Z
Message: Remove unused constants from #25283
Delta:
chunk            stat_size           parsed_size           gzip_size
build               +511 B  (+0.0%)      -2710 B  (-0.2%)     -356 B  (-0.1%)
checkout             -27 B  (-0.0%)        -17 B  (-0.0%)       -8 B  (-0.0%)
checkout~plans        +9 B  (+0.0%)       -148 B  (-0.6%)      -70 B  (-1.1%)
domains              +87 B  (+0.0%)       -181 B  (-0.1%)      -48 B  (-0.1%)
domains~plans        -33 B  (-0.1%)        -30 B  (-0.1%)      -15 B  (-0.3%)
jetpack-connect       +9 B  (+0.0%)       -148 B  (-0.1%)      -74 B  (-0.3%)
manifest              +0 B                  +0 B                +2 B  (+0.0%)
```

## Testing
- Build the app locally. Are there any build warnings like `export 'ACTION_NAME' was not found in 'client/lib/upgrades/action-types.js'`?
- Test the impacted flows. I'm largely unfamiliar with many of these flows and would appreciate details and help testing.

### CODE REVIEW

**Comment by sirreal** [comment]
3c9530f is a garbage commit intended to expose usage of exported action types, and specifically locate unused names (https://github.com/Automattic/wp-calypso/pull/25396#discussion_r194368006).

However, the app seems to build without issue 😕

File: client/lib/upgrades/constants.js [hunk 1]
```diff
@@ -1,79 +0,0 @@
-/** @format */
-
-export const action = {
```
Comment (sirreal) [inline client/lib/upgrades/constants.js]: This file was moved to `action-types`. The single exported object was replaced with named exports, and some were pruned.

**Comment by bsessions85** [comment]
All the e2e tests passed

**Comment by tyxla** [comment]
Uh, seems like this one has been hanging for a while and it now needs a solid rebase. Would you like to take care of that before the review?

**Comment by alisterscott** [comment]
This branch needs a rebase. Are we continuing with this PR?

**Comment by sirreal** [comment]
Rebased. This PR includes some important changes that will make the app more robust and expose some types of errors at build time.

The named exports are also more idiomatic and allow for more optimizations to be applied (tree-shaking).

I'd appreciate reviews and testing before this gets out of date again. The changes are invasive, but not complicated.

**Review by klimeryk — APPROVED** [review]
Tried testing everything I could from domains side of things:

* Adding new G Suite user to a custom domain name.
* Adding new email forward.
* Deleting an email forward.
* Updating nameservers (to/from custom ones).
* Updating DNS records (delete, add, undo delete).
* Using DNS templates to set up email providers.
* Updating WHOIS info.
* Starting and cancelling a transfer out.
* Making the domain primary (including the undo action).
* Verifying the domain.

All good!

And in checkout:

* Modifying the cart (removing items, toggling privacy, etc.).
* Applying a coupon.
* Removing a coupon - this one triggered a fatal error. See this line: https://github.com/Automattic/wp-calypso/pull/25412/files#diff-82722f46268c46a4d427d5927e93609fR134 Looks like you simply forgot to update this one :)

I've looked for any other places that might have slipped through, but looks like the coupon removal was the only one.

So apart from this issue (which is a simple fix), everything else looks good - awesome job! 👏

**Review by aidvu — APPROVED** [review]
Igor was faster. Good job with this! Just don't forget to clean up `removeCoupon()` in `cart.js`.

**Comment by sirreal** [comment]
😱 That is awful! It's the result of some actions that were added after my initial work missed in the rebase (they didn't cause conflicts).

The undefined variables _should_ be caught by eslint, and indeed eslint does report it locally. However, CI does not seem to detect it. Tests are green 😱😵❗️

`eslint` reports 😌:

```
./node_modules/.bin/eslint client/lib/upgrades/actions/cart.js

/Users/jonsurrell/a8c/calypso/client/lib/upgrades/actions/cart.js
  134:9  error  'ActionTypes' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)
```

`eslint-eslines` (used by CI) does not 😱:

```
$ ./node_modules/.bin/eslint-eslines client/lib/upgrades/actions/cart.js
# silence is _not_ golden…
```

@Automattic/team-calypso these are egregious errors missed by CI that folks like myself rely on. It would be good to spend some time investigating this.

**Comment by sirreal** [comment]
Thanks @aidvu and @klimeryk, the issue should be fixed by c4ca45b. Indeed, eslint is happier:

```
$ ./node_modules/.bin/eslint $(git diff --name-only origin/master... | grep -E '^(client|server)' | grep -E '\.jsx?$' )

…/client/my-sites/checkout/cart/secondary-cart.jsx
  28:1  error  componentWillMount is deprecated since React 16.3.0, use UNSAFE_componentWillMount instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillmount  react/no-deprecated

…/client/my-sites/domains/navigation.jsx
  24:1  error  componentWillMount is deprecated since React 16.3.0, use UNSAFE_componentWillMount instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillmount  react/no-deprecated

✖ 2 problems (2 errors, 0 warnings)

```

I'll verify I can add/remove coupons, but would appreciate a quick re-test of that case if you have a moment.

**Comment by aidvu** [comment]
Can confirm it worked for me too.

**Comment by klimeryk** [comment]
Ah, I see - sneaky rebase 😁 Good point about CI not catching this - I'm curious now what's the root issue here. Definitely needs addressing, since like you said, we rely on CI to catch things like this.

**Comment by sirreal** [comment]
#26528 will fix eslint-eslines to report this, so CI shouldn't let problems like this pass anymore 👍

**Pull request merged**